File: backend-nestjs/src/modules/damage-reports/domain/damage-report.repository.ts
import { DamageReport } from './damage-report';

export abstract class DamageReportRepository {
  abstract findAll(): Promise<DamageReport[]>;
  abstract findById(id: string): Promise<DamageReport | null>;
  abstract findByZoneId(zoneId: string): Promise<DamageReport[]>;
  abstract save(report: DamageReport): Promise<DamageReport>;
  abstract remove(id: string): Promise<void>;
}

File: backend-springboot/src/main/java/com/greenalgeria/damagereport/domain/DamageReportRepository.java
package com.greenalgeria.damagereport.domain;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface DamageReportRepository {
    Optional<DamageReport> findById(UUID id);

    List<DamageReport> findAllByOrderByReportedAtDesc();

    List<DamageReport> findByZoneIdOrderByReportedAtDesc(UUID zoneId);

    DamageReport save(DamageReport damageReport);

    boolean existsById(UUID id);

    void deleteById(UUID id);
}
